File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.cpp
class Solution {
public:
    string sortVowels(string s) {
        string vow = "aeiouAEIOU";
        vector<char> vs;

        for(char ch: s){
            if(vow.find(ch) != string :: npos){
                vs.push_back(ch);
            }
        }

        sort(vs.begin(), vs.end());

        int idx=0;
        for(char &ch: s){
            if(vow.find(ch) != string :: npos){
                ch= vs[idx++]; 
            }
        }
        return s;
    }
};

File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vow = 'aeiouAEIOU'
        ins = []
        for ch in s:
            if ch in vow:
                ins.append(ch)
        ins.sort()
        res =[]
        i=0
        for ch in s:
            if ch in vow:
                res.append(ins[i])
                i+=1
            else: res.append(ch)
        return "".join(res)